scanamo/src/main/scala/org/scanamo/generic/SemiautoDerivation.scala

```diff
@@ -18,18 +18,9 @@ package org.scanamo.generic
 
 import magnolia.Magnolia
 import org.scanamo.DynamoFormat
-import scala.language.experimental.macros
 
-final private case class Hidden[A](instance: A) extends AnyVal
+import scala.language.experimental.macros
 
 trait SemiautoDerivation extends Derivation {
-  type Typeclass[A] = HiddenDynamoFormat[A]
```

Comment:
Wasn't sure if I should open another issue about that but this `HiddenDynamoFormat` wasn't actually hidden.
So then this would happen:

```scala
import org.scanamo.DynamoFormat
import org.scanamo.generic.semiauto._

object ScanamoTest extends App {
  class Foo(i: Int, s: String)
  object Foo {
    // This is a self-reference!
    implicit val dynamoFormat: DynamoFormat[Foo] = deriveDynamoFormat
  }

  println(Foo.dynamoFormat) // null
}
```